### DISCUSSION TITLE
Virus protection trips after compilation

### DISCUSSION THREAD

**Discussion by OFreddy** [OpenSalamander/salamander#59] [General]
Are there any options to change the signature of some binaries generated by the compiler? I am facing the problem that some executables are moved to the quarantine during compilation.
I am using Bitdefender on my private computer and some Cisco product in the office. Both have a problem with the compiler output. I guess the scanner report a false positive due to a hash or some faulty heuristic.
Also, some other scanners at virustotal.com are detecting a virus.
This affects the files

- salspawn.exe in Release_x64 and Debug_x64
- salopen.exe in Release_x86 and Release_x64
- zip.spl in Release_x64

Does anyone have an idea how to fix this? Thanks to all for the great work and efforts spend on this project.

![2024-11-29 12_42_37-VirusTotal - File - 512c917cbd0756ecdf6c8d829157a15e81900fca6df8ff7ebf260e940776](https://github.com/user-attachments/assets/1829dd17-cf81-4f8b-9f34-ac389a41e463)

**Comment by melloware** [comment]
I assume they are false positives?  Can you just report them to those virus scanners as false positives.

**Reply by OFreddy** [reply]
Thank you for the reply.
I reported the false positive, but the result is sobering.

So I have changed the code in my fork. The main cause for the false positive is the function EnableExceptionsOn64.

With these changes the Release_X64 Version is accepted by the virus protection.

**Reply by melloware** [reply]
@OFreddy awesome can you submit a PR to fix this for real as other OS users will hit this after the release?

**Reply by melloware** [reply]
@WaldemarH thoughts?

**Reply by OFreddy** [reply]
I feel very unlucky with the current solution. I find it very difficult to explain why it is solved in this way. I will try a few other things. Maybe a better solution can be found.
Unfortunately, I have to reboot after every detection because Bitdefender blocks the file even if you take it out of quarantine.

**Reply by WaldemarH** [reply]
I'm using Bitdefender too and lots of times it thinks that all my VS builds have a virus in them.
I've scanned with different Antivirues and more or less only Bitdefender complains.

So I've removed my whole laboratory folder from any scanning.
